src/azure-cli-core/azure/cli/core/_pkg_util.py

```diff
@@ -0,0 +1,27 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# Each package management system should patch this file with their own implementations of these.
+
+from azure.cli.core._util import COMPONENT_PREFIX
+import azure.cli.core._logging as _logging
+
+logger = _logging.get_az_logger(__name__)
+
+def handle_module_not_installed(module_name):
+    try:
+        import xmlrpclib
+    except ImportError:
+        import xmlrpc.client as xmlrpclib #pylint: disable=import-error
+    query = {'author': 'Microsoft Corporation', 'author_email': 'azpycli'}
+    logger.debug("Checking PyPI for modules using query '%s'", query)
+    client = xmlrpclib.ServerProxy('https://pypi.python.org/pypi')
```

Comment:
It lists the modules from the public server.
This is because:
1. There is no way when you run a command like `az redis` that you'd want it to look in private server as well. For component commands, users explicitly set `--private`.
2. For the `az component list-available` command, it supports Public only because private servers don't have an API like the public PyPI to get available packages. Would have to scrape HTML pages to get these.